### DISCUSSION TITLE
`czg --config=.commitlint/cn.js` doesn't work

### DISCUSSION THREAD

**Discussion by 13OnTheCode** [Zhengqbbb/cz-git#131] [Q&A] [answered]
`pnpm czg --config=.commitlint/cn.js` 不起作用，但是 `pnpm czg --config=.commitlint/cn.json` 是能够正确加载配置文件的，请问 `--config` 是不是仅支持 JSON 文件?

command `pnpm czg --config=.commitlint/cn.js` doesn't work, but `pnpm czg --config=.commitlint/cn.json` correctly loads the configuration file. Does `--config` only support JSON files?

不确定这是一个 bug 还是就是这么设计的，所以没有发起 issues，希望能得到帮助解答

I'm not sure if this is a bug or intended behavior, so I haven't raised any issues. I hope to get some help and clarification.

**Comment by 13OnTheCode** [comment]
emmm 我似乎发现问题所在了，但是依然不清楚是 bug 还是就是这样的，看文档有点不理解。
`czg --config` 仅支持 `czConfig` 配置，比如 `cz.config.js` `.czrc`，不支持 `commitlint.rules`

如果这不是一个 bug 的话希望文档能够完善一下哈哈哈哈，另外 cli 命令能够给予错误提示。

**Comment by 13OnTheCode** [comment]
[https://cz-git.qbb.sh/zh/cli/config#czg-config](https://cz-git.qbb.sh/zh/cli/config#czg-config)
关于 `czg --config` 介绍页面下，虽然说示例给的是指向JSON文件的例子 `czg --config="./config/cz.json"` 
但是并无其他说明，底下的 TIP 告知：关于支持的配置文件以及配置项的更多信息，[→ 查看配置](https://cz-git.qbb.sh/zh/config/)

然后来到配置模板页面，第一项就是 Javascript 模板 `commitlint.config.js`，下面的提示仅仅说明： 如果你的项目未使用 commitlint，并想使用其他配置文件，可以选择下列配置项

可是我明明已经安装 commitlint 依赖了，却还是无法 `pnpm czg --config=.commitlint/commitlint.config.js` 
我觉得这个可能是令人疑惑的地方

**Comment by Zhengqbbb** [comment, marked as answer]
很抱歉，这个文档确实有点粗糙对你造成困扰，稍后我会进行更新

---

> czg --config 仅支持 czConfig 配置，比如 cz.config.js .czrc，不支持 commitlint.rules

是的. `czg --config=` 仅支持 `czConfig` 配置
